python/lsst/ts/ess_sensors/command_handler.py

```diff
@@ -0,0 +1,234 @@
+# This file is part of ts_ess_sensors.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+__all__ = ["CommandHandler"]
+
+import asyncio
+import logging
+import platform
+
+from .ess_instrument_object import EssInstrument
+from .mock.mock_temperature_sensor import MockTemperatureSensor
+from .response_code import ResponseCode
+from .sel_temperature_reader import SelTemperature
+
+
+class CommandHandler:
+    """Handle incoming commands and send replies. Apply configuration and read
+    sensor data.
+
+    Parameters
+    ----------
+    callback: coroutine
+        The callback coroutine used for sending replies, for insatnce for
+        indicating if the command has been executed successfully or not or for
+        sending telemetry.
```

Comment:
I am confused by this description. I think this class represents the sensor end of the TCP/IP connection -- based on the class name and the brief description. If so, I would expect it to:
* Call a callback function when it receives a command.
* Offer a method that the using code calls when it wishes to send a reply.
But it sounds like the opposite.

I suppose I'll figure it out when I read further, but I suspect at the very least that the documentation needs some work, since I won't be the only one confused by this. Possibly the class name, as well.

BTW: you might be interested in the `OneClientServer` in ts_hexrotcomm. It is a wrapper around asyncio's TCP/IP server that enforces just having one client connected (so far it seems all of our controllers work that way) and handles connection, callbacks, and the like.

Reply:
It indeed is intended to do the opposite. The `handle_command` function handles the incoming commands. The `callback` function is used to process the sensor data.  I'll need to update the description to reflect that.
In this case it is a `write` method in the `SocketServer` class. Instead I could pass on the writer of the `SocketServer` but then I'd need to handle exceptions related to the socket connection in the command_handler class and I'd like to keep that separated.

Thanks for mentioning the `OneClientServer` class. I already wondered if something like that exists. Can we extract that to a common project that does not depend on salobj and does not involve DDS traffic? Because that's the purpose of this `ts_ess_sensors` module.

Reply:
That is a good suggestion. Though perhaps for now you could live with the extra dependency? It means installing extra stuff, but you don't have to run it.

Reply:
I don't mind doing that but would it be OK to do that in a separate ticket?

Reply:
I rewrote the doc string.